packages/config/src/load.ts

```diff
@@ -0,0 +1,155 @@
+import { build } from 'esbuild';
+import colors from 'kleur';
+import fs from 'node:fs/promises';
+import path from 'node:path';
+import { pathToFileURL } from 'node:url';
+import {
+  validateConfigHasDefaultExport,
+  validateConfigStructure,
+} from './validate';
+
+type ConfigFileExtension = 'js' | 'mjs' | 'cjs' | 'ts' | 'mts' | 'cts';
+type ConfigFileName = `flatbread.config.${ConfigFileExtension}`;
+
+export const FLATBREAD_CONFIG_FILE_NAMES: ConfigFileName[] = [
+  'flatbread.config.js',
+  'flatbread.config.mjs',
+  'flatbread.config.cjs',
+
+  'flatbread.config.ts',
+  'flatbread.config.mts',
+  'flatbread.config.cts',
+];
+
+const FLATBREAD_CONFIG_FILE_REGEX = /flatbread\.config\.[mc]?[jt]s$/;
+
+const VALID_CONFIG_NAMES_MESSAGE = `Valid config filenames are:\n\t${colors.green(
+  FLATBREAD_CONFIG_FILE_NAMES.join('\n\t')
+)}
+`;
+
+/**
+ * Loads an ESModule-style config file.
+ */
+const esmLoader = async (
+  filename: string,
+  code: string
+): Promise<FlatbreadConfig> => {
+  const configModule = await loadConfigFromBundledFile(filename, code);
+
+  validateConfigStructure(configModule);
+
+  return configModule;
+};
+
+async function loadConfigFromBundledFile(
+  fileName: string,
+  bundledCode: string
+): Promise<FlatbreadConfig> {
+  // swiped a bit from https://github.com/vitejs/vite/blob/main/packages/vite/src/node/config.ts
+  //
+  // for esm, before we can register loaders without requiring users to run node
+  // with --experimental-loader themselves, we have to do a hack here:
+  // write it to disk, load it with native Node ESM, then delete the file.
+  const fileBase = `${fileName}.timestamp-${Date.now()}`;
+  const fileNameTmp = `${fileBase}.mjs`;
+  const fileUrl = `${pathToFileURL(fileBase)}.mjs`;
+  await fs.writeFile(fileNameTmp, bundledCode);
+
+  try {
+    const configModule = await import(fileUrl);
+    validateConfigHasDefaultExport(configModule);
+
+    return configModule.default;
+  } finally {
+    await fs.unlink(fileNameTmp);
+  }
+}
+
+/**
+ * Pulls the user config from an optionally specified filepath.
+ *
+ * By default, this will search the current working directory.
+ *
+ * @param options options for loading the config file, defaults to `{}`. Can pass in `cwd` as a path `string` to override the current working directory.
+ * @returns Promise that resolves to the user config object.
+ */
+export async function loadConfig({ cwd = process.cwd() } = {}): Promise<
+  ConfigResult<FlatbreadConfig>
+> {
+  let configFileName: ConfigFileName;
+  const files = await fs.readdir(cwd);
+
+  const matchingFiles = files.filter((file) =>
+    FLATBREAD_CONFIG_FILE_REGEX.test(file)
+  ) as ConfigFileName[];
+
+  if (matchingFiles.length > 1) {
+    console.error(
+      colors.red(
+        `
+        Multiple config files found (${colors.gray(matchingFiles.join(', '))}).
+        
+        Please declare ${colors.bold('only')} one! 😅\n
+        `
+      ) + VALID_CONFIG_NAMES_MESSAGE
+    );
+    process.exit(1);
```

Comment:
might be a good idea to create an errors file with exported functions that can call to handle this logic
```suggestion
  throw TooManyConfigsError(matchingFiles);
```
```ts
// errors.ts
export class TooManyConfigsError extends Error {
  constructor(matchingFiles) {
    super();
    this.message = colors.red(
      `
      Multiple config files found (${colors.gray(matchingFiles.join(', '))}).
      
      Please declare ${colors.bold('only')} one! 😅\n
      `
    ) + VALID_CONFIG_NAMES_MESSAGE
  }
}

```

Reply:
This is a fantastic suggestion! Added